File: pastebook-db/pastebook-db/Models/Post.cs
using System.ComponentModel.DataAnnotations;

namespace pastebook_db.Models
{
    public class Post
    {
        public int Id { get; set; }
        public string Content { get; set; } = null!;
        public bool IsPublic { get; set; }
        public bool IsEdited { get; set; }
        public DateTime CreatedOn { get; set; }

        public int UserId { get; set; }
        public virtual User User { get; set; } = null!;

        public int? FriendId { get; set; }
        public virtual Friend? Friend { get; set; }

        public ICollection<PostLike>? PostLikeList { get; set; }
        public ICollection<PostComment>? PostCommentList { get; set; }
    }

    public class PostReceiveDTO 
    {
        public string? Content { get; set; }
        public bool IsPublic { get; set; }
        public int UserId { get; set; }
        public int? FriendId { get; set; }
    }

    public class PostDTO
    {
        public int? Id { get; set; }
        public string? Content { get; set; } = null!;
        public bool IsPublic { get; set; }
        public bool IsEdited { get; set; }
        public string CreatedOn { get; set; }

        public int? LikeCount { get; set; }
        public int? CommentCount { get; set; }

        public int UserId { get; set; }
        public int? FriendId { get; set; }

        public ICollection<PostLike>? PostLikeList { get; set; }
        public ICollection<PostComment>? PostCommentList { get; set; }
    }
}

File: postbook-angular/src/app/models/post.ts
import { User } from "./user"

export class Post {
    public id?: string;
    public content?: string;
    public isPublic?: boolean;
    public isEdited?: boolean;
    public createdOn?: Date;

    public likeCount?: number;
    public commentCount?: number;
    public hasLiked?: boolean;

    //FK
    public userId?: string;
    public user?: User;

    public postLikeList: PostLike[] = [];
    public postCommentList: PostComment[] = [];

    public friendId?: string;
}

export class PostLike {
    public id?: string;
    public createdOn?: string;

    public postId?:string;
    public userId?:string;
    public user?: User;
}

export class PostComment {
    public id?: string;
    public comment?: string;
    public createdOn?: string;

    public postId?: string;

    public userId?: string;
    public user?: User;

}
